katdal/sensordata.py

```diff
@@ -887,3 +892,26 @@ def get_with_fallback(self, sensor_type, names):
             except KeyError:
                 logger.debug('Could not find %s sensor with name %r, trying next option' % (sensor_type, name))
         raise KeyError('Could not find any %s sensor, tried %s' % (sensor_type, names))
+
+    # MutableMapping abstract methods
+
+    def __setitem__(self, key, item):
+        with self._lock:
+            self._raw[key] = item
+
+    def __delitem__(self, key):
+        with self._lock:
+            del self._raw[key]
+
+    def __iter__(self):
+        return iter(self._raw)
+
+    def __len__(self):
+        return len(self._raw)
+
+    def __contains__(self, key):
+        # __contains__ is implemented by MutableMapping via __getitem__, but
+        # that does unnecessary extraction. This approach is cheaper but only
+        # reflects keys that have been explicitly created or cached.
```

Comment:
This should be OK because we are already ignoring all the miriad sensors available via katstore...